Mage.Sets/src/mage/cards/i/IndominusRexAlpha.java

```diff
@@ -0,0 +1,183 @@
+                + ". The same is true for " + getOtherChoicesAsString() + ".";
+        for (CounterType counterType : copyableCounters) {
+            Counter counter = counterType.createInstance();
+            if (!(counter instanceof AbilityCounter)) {
+                throw new IllegalArgumentException(counterType + " is not an ability counter");
+            }
+        }
+    }
+
+    protected String getFirstChoiceAsString(){
+        return copyableCounters.toArray(new CounterType[0])[0].getName();
+    }
+
+    protected String getOtherChoicesAsString(){
+        StringBuilder sb = new StringBuilder();
+        CounterType[] choicesArray = copyableCounters.toArray(new CounterType[0]);
+
+        // Skip first counter name - flying
+        for (int i = 1; i < copyableCounters.size(); i++){
+            String separator = ", ";
+            if (i == copyableCounters.size()-2){
+                separator = ", and ";
+            } else if (i == copyableCounters.size()-1){
+                separator = "";
+            }
+            sb.append(choicesArray[i].getName()).append(separator);
+        }
+        return sb.toString();
+    }
+
+    private IndominusRexAlphaCountersEffect(final IndominusRexAlphaCountersEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public IndominusRexAlphaCountersEffect copy() {
+        return new IndominusRexAlphaCountersEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+
+        // EntersBattlefieldAbility is static, see AddCountersSourceEffect
+        Permanent permanent = game.getPermanentEntering(source.getSourceId());
+        if (permanent == null) {
+            return true;
+        }
+
+        TargetCard target = new TargetDiscard(0, Integer.MAX_VALUE, StaticFilters.FILTER_CARD_CREATURE, controller.getId());
+        controller.choose(outcome, controller.getHand(), target, source, game);
+        List<UUID> chosenTargets = target.getTargets();
+
+        // the basic event is the EntersBattlefieldEvent, so use already applied replacement effects from that event
+        List<UUID> appliedEffects = (ArrayList<UUID>) this.getValue("appliedEffects");
+
+        boolean addedCounters = false;
+
+        for (CounterType counterType : copyableCounters) {
+
+            Counter counter = counterType.createInstance();
+            Ability abilityToCopy = ((AbilityCounter)(counter)).getAbility();
+            Class<? extends Ability> abilityClass = abilityToCopy.getClass();
+
+            boolean addedCounterType = false;
+
+            for (UUID targetId : chosenTargets) {
+
+                Card card = game.getCard(targetId);
+                if (card == null){
+                    continue;
+                }
+
+                for (Ability ability : card.getAbilities()) {
```

Comment:
Must use card.getAbilities(game) - it will return dynamically gained abilities too.

Fast hint: try to use game and source param as much as possible (if you see multiple methods then use version with game/source if possible).